### PR TITLE
Added support for publishing this library to maven central

### PR DESCRIPTION
**Pull request by christianrowlands**
- Also updated the gradle version and gradle wrapper
- Updated the README with the GAV coordinates in maven central

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,19 @@ The gRPC website has tutorials for utilizing gRPC + protobuf for [Android](https
 [Java](https://grpc.io/docs/quickstart/java/).
 
 
+Getting Started
+-----------------------------------
+#####Add Network Survey Messaging to your project
```
Comment (christianrowlands) [inline README.md]: The formatting is messed up on this.  I will fix it.

File: .gitignore [hunk 2]
```diff
@@ -5,6 +5,11 @@
...
 # Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 
+# All of .idea
+.idea
+.project
+.settings
+
 # User-specific stuff
 .idea/**/workspace.xml
 .idea/**/tasks.xml
```
Comment (lstreepy) [inline .gitignore]: Why are these still necessary if you exclude all of .idea above

File: build.gradle [hunk 3]
```diff
@@ -68,6 +80,24 @@ clean {
...
 }
 
+task sourceJar(type: Jar) {
+    classifier "sources"
+    from sourceSets.main.allJava
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier "javadoc"
+    from javadoc.destinationDir
```
Comment (lstreepy) [inline build.gradle]: Can just say 'from javadoc' here

File: build.gradle [hunk 4]
```diff
@@ -81,10 +111,69 @@ clean {
     return path
 }*/
 
-publishing {
-    publications {
-        mavenJava(MavenPublication) {
-            artifact jar
+// Configure publishing to the Sonatype Maven Central repo
+uploadArchives {
```
Comment (lstreepy) [inline build.gradle]: 'uploadArchives' comes from the 'maven' plugin, and is ancient and deprecated.  You should be using the 'publishing' configuration which is provided by the 'maven-publish' plugin.

File: build.gradle [hunk 5]
```diff
@@ -81,10 +111,69 @@ clean {
...
         }
     }
+}
+
+/**
+ * Try to locate a credential value (user name or password) in the environment,
+ * or falling back to the credentials store.
+ *
+ * This supports building locally on a developers machine and on a Jenkins server with credentials bound
+ * into the environment.
```
Comment (lstreepy) [inline build.gradle]: Probably should document the pattern for the environment variable name,

File: build.gradle [hunk 6]
```diff
@@ -81,10 +111,69 @@ clean {
...
+ *
+ * @param value The suffix to use for the credential lookup key.  It will be appended to "sonatype".
+ * @return The credential value if located.
+ */
+private String findCredentialValue(String value) {
+    def cred = System.getenv("${name.toUpperCase()}_NEXUS_${value.toUpperCase()}")
```
Comment (lstreepy) [inline build.gradle]: what 'name' is being referenced in 'name.toUpperCase' here?

File: build.gradle (hunk 4 above)
Comment (christianrowlands) [reply build.gradle]: Hmm, strange that Maven Central uses uploadArchives in their tutorial:  https://central.sonatype.org/pages/gradle.html

I see the gradle documentation on the new approach so I will update it to use that:  https://docs.gradle.org/current/userguide/publishing_setup.html#publishing_overview

File: build.gradle (hunk 5 above)
Comment (christianrowlands) [reply build.gradle]: I am going to just remove it.  It is not necessary at this point.

File: build.gradle (hunk 6 above)
Comment (christianrowlands) [reply build.gradle]: Copy and paste error 😬

File: build.gradle (hunk 4 above)
Comment (christianrowlands) [reply build.gradle]: Ok, how does it look now?

File: README.md [hunk 7]
```diff
@@ -28,12 +28,25 @@ The gRPC website has tutorials for utilizing gRPC + protobuf for [Android](https
...
+dependencies {
+    implementation 'com.craxiom:network-survey-messaging:0.0.1'
+}
+```
+
+
 Build and development instructions
 -----------------------------------
-###Building the project
+### Building the project
```
Comment (christianrowlands) [inline README.md]: I need to delete this duplicate line.

**Comment by christianrowlands** [comment]
I have tested it, and it uploads everything.  It also passes all the Maven Central checks.  However, I have not pulled it down from maven central and verified that it works in a project.

I figured just the releases was good enough.

**Comment by christianrowlands** [comment]
Ok, I think I have it now.  Hopefully this is good.

**Pull request merged**